lib/liberic/sdk/api.rb

```diff
@@ -2,11 +2,20 @@ module Liberic
   module SDK
     module API
       extend FFI::Library
-      ffi_lib Liberic.library_path
 
-      def self.attach_eric_function(name, params, return_type, original_name = nil)
-        original_name ||= 'Eric' + name.to_s.capitalize.gsub(/(?:_|(\/))([a-z\d]*)/) { "#{$1}#{$2.capitalize}" }
-        attach_function(name, original_name, params, return_type)
+      if Liberic.binaries_unavailable?
+        # define fake functions to allow tests pass
+        def self.attach_eric_function(name, params, return_type, original_name = nil)
+          define_method(name) { |*options| true }
+        end
+        extend self
```

Comment:
Just to understand - why you need to `extend self`?

Reply:
It defines an instance method in the receiver https://apidock.com/ruby/Module/define_method
Here we are using Module so the "extend self" keeps the instance methods public.

Anyway I changed this code. Now I will use [define_singleton_method ](https://apidock.com/ruby/Object/define_singleton_method) that doesn't require "extend self"